cognee/infrastructure/databases/graph/neptune_driver/adapter.py

```diff
@@ -819,6 +819,89 @@ async def get_edges(self, node_id: str) -> List[EdgeData]:
             logger.error(f"Failed to get edges for node {node_id}: {error_msg}")
             raise Exception(f"Failed to get edges: {error_msg}")
 
+    async def get_disconnected_nodes(self) -> list[str]:
+        """
+        Find and return nodes that are not connected to any other nodes in the graph.
+
+        Returns:
+        --------
+
+            - list[str]: A list of IDs of disconnected nodes.
+        """
+        query = f"""
+            MATCH(n :{self._GRAPH_NODE_LABEL})
+            WHERE NOT (n)--()
+            RETURN COLLECT(ID(n)) as ids
+        """
+
+        results = await self.query(query)
+        return results[0]["ids"] if len(results) > 0 else []
+
+    async def get_predecessors(self, node_id: str, edge_label: str = None) -> list[str]:
+        """
+        Retrieve the predecessor nodes of a specified node based on an optional edge label.
+
+        Parameters:
+        -----------
+
+            - node_id (str): The ID of the node whose predecessors are to be retrieved.
+            - edge_label (str): Optional edge label to filter predecessors. (default None)
+
+        Returns:
+        --------
+
+            - list[str]: A list of predecessor node IDs.
+        """
+
+        edge_label = f" :{edge_label}" if edge_label is not None else ""
+        query = f"""
+        MATCH (node)<-[r{edge_label}]-(predecessor)
+        WHERE node.id = $node_id
+        RETURN predecessor
+        """
+
+        results = await self.query(
+            query,
+            dict(
+                node_id=node_id,
+            ),
+        )
+
+        return [result["predecessor"] for result in results]
+
+    async def get_successors(self, node_id: str, edge_label: str = None) -> list[str]:
```

Comment:
```suggestion
    async def get_successors(self, node_id: str, edge_label: str = "") -> list[str]:
```
nit: I think this is better